modules/imaging_uploader/ajax/read_log.php

```diff
@@ -12,96 +12,71 @@
  * @link     https://github.com/aces/Loris-Trunk
 */
 
-$user =& User::singleton();
-if (!$user->hasPermission('mri_upload')) {
-    header("HTTP/1.1 403 Forbidden");
-    exit;
+if (!\User::singleton()->hasPermission('imaging_uploader')) {
+    http_response_code(403);
+    return;
 }
 
 header("Cache-Control: no-store, no-cache, must-revalidate");
 require_once "../tools/generic_includes.php";
 
 
-///////////////////////////////////////////////////////////////////////////////
-//////////////////////////////get the log file-name////////////////////////////
-///////////////////////////////////////////////////////////////////////////////
+// Get the path to the file most recently modified by the MRI upload process.
 $data_source_file = getLatestLogFile();
 
-/**
- * Create a temp file to avoid the original file to be truncated
- */
+// Create a temp file to avoid truncating the original file.
 $temp_data_source_file = $data_source_file . ".tmp";
 
-///file doesn't exist create it
+// Create new temp file for reading if it doesn't exist.
 if (file_exists($temp_data_source_file)) {
-    if (!file_exists($temp_data_source_file)) {
-        if (!copy($data_source_file, $temp_data_source_file)) {
-            print "cannot be copied";
-        }
+    if (!copy($data_source_file, $temp_data_source_file)) {
+        throw new \LorisException(
+            "Could not copy " .
+            htmlspecialchars($data_source_file, ENT_QUOTES) .
+            " to " .
+            htmlspecialchars($temp_data_source_file, ENT_QUOTES)
+        );
     }
 }
 
-
 clearstatcache();
-
-///create a new temp file for the info
-
-
-
-$file = fopen($temp_data_source_file, "r+");
-
-////if the file is opened
-if ($file) {
-    ////////Lock the file
-    if (flock($file, LOCK_EX)) {
-
-        //get the data from the file
-        $data = fread($file, filesize($temp_data_source_file));
-        ftruncate($file, 0);//truncate the file
-        fflush($file); //writes the buffered output to the file
-        flock($file, LOCK_UN);  //locks the file
-        echo $data;
-    } else {
-        echo "could not lock file";
-    }
+$contents = file_get_contents($temp_data_source_file);
+if (!$contents) {
```

Comment:
Doesn't remove the temp file. This will fill the disk rapidly with unneeded file
Is it still needed with this new algorithm?

Reply:
I don't know whether it was needed in the first place. Perhaps it would be better to remove the temp file entirely and simply read the original?

Reply:
Probably yes, I can't see a reason to make a copy and load the copy.
Old code was actually emptying the content of the files (without remove it)